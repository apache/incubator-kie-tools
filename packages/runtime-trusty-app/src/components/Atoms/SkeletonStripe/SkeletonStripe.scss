.skeleton {
  &__stripe {
    width: 50%;
    height: 1.5em;
    display: block;
    background: #f6f7f8;
    background: linear-gradient(to right, #f5f5f5 8%, #eeeeee 18%, #f5f5f5 33%);
    background-size: 800px 104px;
    animation: linear 1.3s placeHolderShimmer infinite;
    border-radius: 4px;
    &--md {
      width: 80%;
    }
    &--lg {
      width: 60%;
    }
    &--inline {
      display: inline-block;
      vertical-align: middle;
      font-size: 1em;
      line-height: inherit;
      height: 1em;
      width: 200px;
      margin: 0 10px;
    }
  }
}

@keyframes placeHolderShimmer {
  0% {
    background-position: -468px 0;
  }
  100% {
    background-position: 468px 0;
  }
}
