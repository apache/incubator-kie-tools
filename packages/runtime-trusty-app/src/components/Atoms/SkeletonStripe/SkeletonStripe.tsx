/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React, { useMemo } from "react";
import "./SkeletonStripe.scss";

type SkeletonStripeProps = {
  customStyle?: React.CSSProperties;
  isInline?: boolean;
  size?: "sm" | "md" | "lg";
};

const SkeletonStripe = (props: SkeletonStripeProps) => {
  const { isInline = false, size = "sm", customStyle } = props;
  const stripeStyle = useMemo(() => customStyle || {}, [customStyle]);
  let cssClasses = "skeleton__stripe";

  if (isInline) {
    cssClasses += " skeleton__stripe--inline";
  }
  if (size !== "sm") {
    cssClasses += ` skeleton__stripe--${size}`;
  }

  return <span className={cssClasses} style={stripeStyle} />;
};

export default SkeletonStripe;
