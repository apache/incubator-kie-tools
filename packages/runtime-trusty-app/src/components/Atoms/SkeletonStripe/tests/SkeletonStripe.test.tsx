/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from "react";
import { shallow } from "enzyme";
import SkeletonStripe from "../SkeletonStripe";

describe("SkeletonStripe", () => {
  test("renders a small stripe", () => {
    const wrapper = shallow(<SkeletonStripe />);
    expect(wrapper.find("span").props().className).toMatch("skeleton__stripe");
    expect(wrapper).toMatchSnapshot();
  });

  test('renders a medium size stripe when size "md" is passed', () => {
    const wrapper = shallow(<SkeletonStripe size="md" />);
    expect(wrapper.find("span").props().className).toMatch("skeleton__stripe skeleton__stripe--md");
  });
  test('renders a large size stripe when size "lg" is passed', () => {
    const wrapper = shallow(<SkeletonStripe size="lg" />);
    expect(wrapper.find("span").props().className).toMatch("skeleton__stripe skeleton__stripe--lg");
  });
  test("renders an inline stripe when isInline is passed", () => {
    const wrapper = shallow(<SkeletonStripe isInline />);
    expect(wrapper.find("span").props().className).toMatch("skeleton__stripe skeleton__stripe--inline");
  });
  test("renders an inline stripe with custom styles", () => {
    const wrapper = shallow(<SkeletonStripe customStyle={{ width: 600 }} />);
    expect(wrapper.find("span").props().style).toStrictEqual({ width: 600 });
  });
});
