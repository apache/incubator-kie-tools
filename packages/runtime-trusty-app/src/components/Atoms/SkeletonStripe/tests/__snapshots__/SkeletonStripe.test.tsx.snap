// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonStripe renders a small stripe 1`] = `
<span
  className="skeleton__stripe"
  style={Object {}}
/>
`;
